darts/tests/test_torch_forecasting_model.py

```diff
@@ -0,0 +1,142 @@
+            # no exception is raised
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series, epochs=1)
+
+            with self.assertWarnsRegex(UserWarning, "You already have model data for the '{}' name and you "
+                                                    "initialized it with".format(model_name)):
+                model2 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name, force=True)
+
+            with self.assertRaisesRegex(ValueError, "You forced initialization of the model but.*"):
+                model2.fit(series, epochs=1)
+
+        def test_create_instance_existing_model_with_name_force_fit_with_reset(self):
+            model_name = 'test_model'
+            model1 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name)
+            # no exception is raised
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series, epochs=1)
+
+            model2 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name, force=True)
+            model2.reset_model()
+            model2.fit(series, epochs=1)
+            # no exception is raised
+
+        # n_epochs=20, fit|epochs=None, total_epochs=0 - train for 20 epochs
+        def test_train_from_0_to_20_no_fit_epochs(self):
+            model1 = RNNModel(10, 10, n_epochs=20, work_dir=self.temp_work_dir)
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series)
+
+            self.assertEqual(model1.total_epochs, 20)
+
+        # n_epochs = 20, fit|epochs=None, total_epochs=20 - train for another 20 epochs
+        def test_train_from_20_to_40_no_fit_epochs(self):
+            model1 = RNNModel(10, 10, n_epochs=20, work_dir=self.temp_work_dir)
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series)
+            self.assertEqual(model1.total_epochs, 20)
+
+            model1.fit(series)
+            self.assertEqual(model1.total_epochs, 40)
+
+        # n_epochs = 20, fit|epochs=None, total_epochs=10 - train for another 20 epochs
+        def test_train_from_10_to_20_no_fit_epochs(self):
+            model1 = RNNModel(10, 10, n_epochs=20, work_dir=self.temp_work_dir)
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series, epochs=10)  # cheating here a little bit
```

Comment:
why cheating?

Reply:
it's cheating, because the case is that user interrupted training with Ctrl-C after 10 epochs and this code simulates it. But I can ofc remove the comment.

Reply:
Could you maybe write that in a comment? Something like 'this line creates a model in the same state as it would be in if training had been interrupted after 10 epochs'. At least for me it would clarifies things.

Reply:
Sure.